graylog2-web-interface/src/views/components/contexts/FieldTypesContext.jsx

```diff
@@ -0,0 +1,8 @@
+// @flow strict
+import * as React from 'react';
+
+import { singleton } from 'views/logic/singleton';
+import type { FieldTypesStoreState } from 'views/stores/FieldTypesStore';
+
+const FieldTypesContext = React.createContext<?FieldTypesStoreState>();
```

Comment:
Instead of reusing the store state, we should create an own type in the context that we are populating. If it is structurally identical to `FieldTypesStoreState`, we do not even need a conversion. This helps us decouple the context from any component that is providing it.